test/e2e/storage/pvc_protection.go

```diff
@@ -0,0 +1,134 @@
+	})
+
+	AfterEach(func() {
+		framework.ExpectNoError(client.StorageV1().StorageClasses().Delete(storageClass.Name, metav1.NewDeleteOptions(0)))
+	})
+
+	It("Verify \"immediate\" deletion of a PVC that is not in active use by a pod", func() {
+		By("Deleting the PVC")
+		err = client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(pvc.Name, metav1.NewDeleteOptions(0))
+		Expect(err).NotTo(HaveOccurred(), "Error deleting PVC")
+		waitForPersistentVolumeClaimBeRemoved(client, pvc.Namespace, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout)
+	})
+
+	It("Verify that PVC in active use by a pod is not removed immediatelly", func() {
+		By("Creating a Pod that becomes Running and therefore is actively using the PVC")
+		pvcClaims := []*v1.PersistentVolumeClaim{pvc}
+		pod, err := framework.CreatePod(client, nameSpace, nil, pvcClaims, false, "")
+		Expect(err).NotTo(HaveOccurred(), "While creating pod that uses the PVC or waiting for the Pod to become Running")
+
+		By("Deleting the PVC, however, the PVC must not be removed from the system as it's in active use by a pod")
+		err = client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(pvc.Name, metav1.NewDeleteOptions(0))
+		Expect(err).NotTo(HaveOccurred(), "Error deleting PVC")
+
+		By("Checking that the PVC status is Terminating")
+		pvc, err = client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(pvc.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "While checking PVC status")
+		Expect(pvc.ObjectMeta.DeletionTimestamp).NotTo(Equal(nil))
+
+		By("Deleting the pod that uses the PVC")
+		err = framework.DeletePodWithWait(f, client, pod)
+		Expect(err).NotTo(HaveOccurred(), "Error terminating and deleting pod")
+
+		By("Checking that the PVC is automativally removed from the system because it's no longer in active use by a pod")
+		waitForPersistentVolumeClaimBeRemoved(client, pvc.Namespace, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout)
+	})
+})
+
+// waitForPersistentVolumeClaimBeRemoved waits for a PersistentVolumeClaim to be removed from the system until timeout occurs, whichever comes first.
+func waitForPersistentVolumeClaimBeRemoved(c clientset.Interface, ns string, pvcName string, Poll, timeout time.Duration) error {
+	framework.Logf("Waiting up to %v for PersistentVolumeClaim %s to be removed", timeout, pvcName)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(Poll) {
+		_, err := c.CoreV1().PersistentVolumeClaims(ns).Get(pvcName, metav1.GetOptions{})
+		if err != nil {
+			if apierrs.IsNotFound(err) {
+				framework.Logf("Claim %q in namespace %q doesn't exist in the system", pvcName, ns)
+				return nil
+			}
+			framework.Logf("Failed to get claim %q in namespace %q, retrying in %v. Error: %v", pvcName, ns, Poll, err)
+		}
+	}
+	return fmt.Errorf("PersistentVolumeClaim %s is not removed from the system within %v", pvcName, timeout)
+}
+
+func skipUnlessPVCProtectionEnabled() {
```

Comment:
Does this work externally in a already running cluster? What I understand is - once a cluster is already running with feature flag enabled, the e2e test obviously runs externally and the e2e process does not know, if the feature flag is enabled or not. May be there is some magic that makes it work with e2e job but it does not work locally for me. 

I also noticed that some of the tests that @msau42  introduced uses similar feature gating in e2e tests.

Reply:
I "copied" the feature gating from @msau42 

The feature gating doesn't work externally when I try it on my notebook. I have no idea if it works in e2e jobs.

Reply:
I don't recall having feature gating in my e2e tests.  This won't work as there's no way to externally check if a feature gate is enabled in a cluster.

The way to add e2e tests for alpha features is to:
* Add a `[Feature: blah]` tag to your suite
* In the test infra config, update the alpha job to include your tag added above.

Reply:
@msau42 @gnufied thank you for advice. I removed the feature gating.